consensus/src/consensus.rs

```diff
@@ -223,18 +279,33 @@ impl<R: ConsensusRpc> ConsensusClient<R> {
                 }
             }
         }
-        self.store_finality_update(&finality_update, true).await?;
+        self.store_updates_until_finality_update(&finality_update)
+            .await?;
 
         Ok(())
     }
 
-    async fn bootstrap(&mut self) -> Result<()> {
+    async fn bootstrap(&mut self, base_slot: u64) -> Result<()> {
+        if let Some(stored_base_slot) = self.storage().get_base_beacon_header_slot()? {
+            if stored_base_slot != base_slot {
```

Comment:
> so here it means on-chain and native base slot must be the same, not even the condition that `stored_base_slot < base_slot`?

Yep.

### Conclusion

If the tip slot in storage is not the same as the tip slot in the CKB chain, we can rollback some blocks or wait for more blocks.

But if the base slots are not the same, I think users have to check if the light client cell they set is correct.

### Reason

Different base slots means total different MMR data in storage.

If there are already a lot of MMR data in storage, delete and re-create them will cost a lot of CPUs and time: it's expensive.

All expensive operations should be confirmed from users before they are performed.
Since that, I think this should NOT be an automatic operation.

Reply:
make sense, I should correct it in the Relayer repo as well.